core/src/com/biglybt/core/util/NetUtils.java

```diff
@@ -549,37 +547,15 @@ at whatevercalled getNetworkInterfaces()
 			se = e;
 		}
 
-		// Java 7 has getByIndex
-		try {
-			Method mGetByIndex = NetworkInterface.class.getDeclaredMethod(
-					"getByIndex", int.class);
-			List<NetworkInterface> list = new ArrayList<>();
-			int i = 0;
-			do {
-				//NetworkInterface nif = NetworkInterface.getByIndex(i);
-				NetworkInterface nif = null;
-				try {
-					nif = (NetworkInterface) mGetByIndex.invoke(null, i);
-				} catch (IllegalAccessException e) {
-					break;
-				} catch (InvocationTargetException ignore) {
-					// getByIndex throws SocketException
-				}
-				if (nif != null) {
-					list.add(nif);
-				} else if (i > 0) {
-					break;
-				}
-				i++;
-			} while (true);
-			if (list.size() > 0) {
-				return Collections.enumeration(list);
-			}
-		} catch (NoSuchMethodException ignore) {
+		final List<NetworkInterface> list = new ArrayList<>();
+
+		collectNetworkInterfacesByIndex(list);
+
+		if (!list.isEmpty()) {
+			return Collections.enumeration(list);
```

Comment:
This "Java 7 preview" block of code made the runtime return at this line when running Java 7 or higher.

I believe the next block was for JDK 6 and earlier only, making that a tautology on the current baseline. 
Should I go ahead and remove it ?

Reply:
I mean, it only returned here **if** the list was non-empty. Which doesn't seem like it's _guaranteed_ even still, so I'd say it's far from tautological.

**Most** of the time the code will never get this far, period. It'll return the valid list produced by `getNetworkInterfaces()` on line 526 (524 in your PR), none of the code past that will even execute, and all will be right with the world.

But on some system where `getNetworkInterfaces()` _fails_, can we say for sure that `getByIndex()` won't fail as well? Especially when the code is just _guessing_ indices.

It's clear from the docs that the `getByIndex()` method was **meant** to be used by code that _already_ had a known valid index, most likely retrieved by previously calling `.getIndex()` on an interface object and storing it somewhere. And the docs explicitly refuse to guarantee anything about the index values themselves, other than that they're integers >= 0. So there's nothing stopping some implementation from using high and/or randomized index values that the `getByIndex()` approach won't ever find.

In which case, guessing names _could_ theoretically still work as an absolute last-resort fallback.

Reply:
> So there's nothing stopping some implementation from using high and/or randomized index values that the `getByIndex()` approach won't ever find.

Oh, nope, wait, I'm wrong about that, sorry. Since the code will (in theory) try every integer if `.getByIndex()` is available, the only way that would fail to eventually find **something** is if `.getByIndex()` was implemented but couldn't actually be used to look up interfaces, which they're not allowed to do.

In that case, I don't see any reason the name-guessing code would still be needed.